packages/framework/src/Model/Product/ProductVisibilityRepository.php

```diff
@@ -278,12 +286,13 @@ protected function hideVariantsWithInvisibleMainVariant($onlyMarkedProducts)
                         AND mpv.visible = FALSE
                 )
             ' . $onlyMarkedProductsCondition,
-            new ResultSetMapping()
+            [
+                'variantTypeVariant' => Product::VARIANT_TYPE_VARIANT,
```

Comment:
Maybe this static value should not be passed by parameter as well.

Reply:
I prefer to keep as the parameter as it's bound to product and usage of the value can be easily found.